Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -155,6 +155,12 @@ public override void Analyze(AnalyzeContext context)
             {
                 try
                 {
+                    var fileInfo = new FileInfo(filePath);
+                    if (context.FileSizeInKilobytes != -1 && fileInfo.Length / 1024 > context.FileSizeInKilobytes)
```

Comment:
@marmegh A comment here would help. What is the logic? I am guessing `context.FileSizeInKilobytes` is the size of the file under analysis. What does it mean for the value to be -1? What is `fileInfo.Length`?

Reply:
FileSizeInKilobytes is not a great name. It is really a limit or threshold. For the record, I probably gave this argument a bad name.

Here's a comment that I think addresses

// Ensure that the byte of the file does not exceed the limit set by the
// `file-size-in-kilobytes` argument (which will be -1 if not set, in which
// case all files should be analyzed no matter their size).